lib/PreviewProvider.js

```diff
@@ -110,6 +110,17 @@ PreviewProvider.prototype = {
   },
 
   /**
+    * A decodeURIComponent that doesn't throw on invalid URI components
+    */
+  _safeDecodeURIComponent(str) {
+    try {
+      return decodeURIComponent(str);
+    } catch (e) {
+      return str;
```

Comment:
AFAICS `decodeURIComponent` fails when there's an invalid URL passed.

Is there a situation where it is failing on valid URL's?

For instance, this passes: `decodeURIComponent("foo://example.com/bar?baz=quux&foo=bar%20baz")`

But this doesn't: `decodeURIComponent("foo://example.com/bar?baz=quux&foo=bar%baz")`

This is rightly so, because the URL is invalid